lib/membrane/parent_spec.ex

```diff
@@ -343,14 +343,17 @@ defmodule Membrane.ParentSpec do
   - `toilet_capacity` - Used when a toilet is created, that is for pull input pads that have push output pads
     linked to them. When a push output produces more buffers than the pull input can consume, the buffers are accumulated
     in a queue called toilet. If the toilet size grows above its capacity, it overflows by raising an error.
-  - `demand_excess` - Used only for pads working in pull mode with manual demands. See `t:Membrane.Pad.mode_t/0`
-    and `t:Membrane.Pad.demand_mode_t/0` for more info.
-  - `min_demand_factor` - A factor used to calculate minimal demand (`minimal_demand = demand_excess * min_demand_factor`)
-    by which `demand_excess` is multiplied by to calculate the minimal demand. Used only for pads working in pull
-    mode with manual demands. See `t:Membrane.Pad.mode_t/0` and `t:Membrane.Pad.demand_mode_t/0` for more info. Defaults
-    to `#{Membrane.Core.Element.InputQueue.default_min_demand_factor()}` (the default may change in the future).
-  - `auto_demand_size` - Used only for pads working in pull mode with automatic demands. See `t:Membrane.Pad.mode_t/0`
+  - `demand_excess` - Demand that will be generated automatically by Membrane to allow smooth, concurrent processing.
+    All buffers received in response to that demand will be queued internally until they are actually demanded by user.
+    Used only for pads working in pull mode with manual demands. See `t:Membrane.Pad.mode_t/0`
```

Comment:
This still doesn't explain everything and only remembering the old `preffered_size` I could get it. I'd modify the docs and change the name of the option:
```suggestion
  - `initial_demand` - The initial demand sent to the linked output pad. It is used to fill the queue of the input pad.
    That allows for fulfilling the demands of the element by taking data from the queue while the actual sending of demands is done asynchronously, smoothing the processing.
    Used only for pads working in pull mode with manual demands. See `t:Membrane.Pad.mode_t/0`
```

Ideas for alternative names:
* `warm_up_demand`
* `starting_demand`

This may however affect the way `min_demand_factor` is defined, I'm not sure if if makes sense to define the `min_demand` as function of this demand and calculated using a factor

Reply:
`initial demand` seems like it was only sent initially and didn't matter afterwards. I refactored the description but left `demand_excess`